// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// recall.proto
syntax = "proto3";
import "user_info.proto";
package recall;
option cc_generic_services = false;

message RecallRequest{
  string log_id = 1;
  user_info.UserInfo user_info = 2;
  string recall_type= 3;
  uint32 request_num= 4;
}

message RecallResponse{
   message Error {
        uint32 code = 1; 
        string text = 2;
   }
   message ScorePair {
        string nid = 1;
        float score = 2;
    };
   Error error = 1;
    repeated ScorePair score_pairs = 2;
}

service RecallService {
    rpc recall(RecallRequest) returns (RecallResponse);
};

